sdks/python/apache_beam/runners/portability/portable_runner_test.py

```diff
@@ -201,7 +204,8 @@ class PortableRunnerTestWithExternalEnv(PortableRunnerTest):
   @classmethod
   def setUpClass(cls):
     cls._worker_address, cls._worker_server = (
-        worker_pool_main.BeamFnExternalWorkerPoolServicer.start())
+        worker_pool_main.BeamFnExternalWorkerPoolServicer.start(
```

Comment:
Logically this is duplicates the experiment flag above. I can see why that is currently necessary, but maybe it would be better to provide the pipeline options to the worker pool servicer? (Inside the container this would happen via provisioning endpoint and environment.)

Reply:
Will look into this.

Reply:
Since this affects multiple parameters of the `start` method (e.g. `worker_threads`), I would handle this in a follow-up. It looks like either way, via individual parameters or via pipeline options here, there will be duplicate code paths, due to this method being used in a test context and for the main entry point in `if __name__ == '__main__'`. To fix this it would require restructuring of the entire file, which I see out of scope for this PR. But good to think about doing that eventually.